Add Contributors to README using All-Contributors

thedavidprice opened the pull request:
I used https://allcontributors.org/ CLI (via custom script) to batch populate the README with all current contributors. (Note: the diff between GitHub count and README count is bots.) Via enabling a Bot for this project, we can manually trigger the addition of new contributors via an automated PR. It's not as seamless as I'd like, but it's also not terribly cumbersome -- example usage here: https://allcontributors.org/docs/en/bot/usage

Questions:
- do we like the output and layout at the bottom of the README.md?
- should we keep the counter?
- overall, is this an acceptable process and tool? (I tried a few others, including GH Actions automation, and they came up wanting)

Decisions:
- [ ] This README is included in the redwoodjs.com docs. Do we need special handling for this new Contributors section?
- [ ] Should we also do this for Redwoodjs.com repo?

TO DO
- [ ] Squash this monstrosity before merging
- [ ] Document the CLI 'check' and 'generate' process (and determine whether or not to add package)
- [ ] Confirm Bot is enabled/configured correctly and test

jeliasson commented:
Great stuff @thedavidprice, and thanks for taking the time to add more recognizement! 🚀

> * do we like the output and layout at the bottom of the [README.md](https://github.com/redwoodjs/redwood/blob/92db0c36f5bbe5dcd144b7609378d5709f087656/README.md#contributors)? 

Yes, but I suggest removing the 'code' icon if there is no plans of adding some manual distinguishment in the `contributions` array. Considering the amount of [contribution types](https://allcontributors.org/docs/en/emoji-key) the bot supports, it might end up being cluttered. Maybe?

> * should we keep the counter?

I say remove it; less clutter. For anyone curious, the actual up-to-date contributor list can be seen on a per repo basis 😊

> * overall, is this an acceptable process and tool? (I tried a few others, including GH Actions automation, and they came up wanting)

I think this is a great starting point. GH Actions Automation, pulling contributions for all repos under [redwoodjs](https://github.com/redwoodjs) might be ideal in the future, but that's room for improvements.

Tobbe commented:
![image](https://user-images.githubusercontent.com/30793/88176995-6bf1fa80-cc28-11ea-894f-e39970742c1d.png)

The badge looks to be clickable, but nothing actually happens when you click it

> should we keep the counter?

You mean the counter at the end of the badge? If we're keeping the badge, I think we can keep the counter as well. If we're removing it all together, then I don't think there's any need to keep the counter.

I'd love to have all redwoodjs.com contributors in this list as well. The docs and the tutorial are super important, and we want to acknowledge anyone who helps make them even better

mojombo commented:
Thanks for working this up!

* I like the list at the bottom of the README.
* Seems fine to remove the "count" button. It's weird that it seems to be a link but doesn't do anything, and the total number isn't really the point.
* I think this should be a place where we recognize contributions across all the main Redwood repos, including redwoodjs.com and CRA. Perhaps we could have a different emoji to show for contributors depending on the repo? Or maybe that doesn't really matter right now and we can just leave them off entirely as @jeliasson suggests.

thedavidprice commented:
Ok, I've taken the comments here into consideration and made these changes:
- removed the counter
- removed the emojis

I've also updated the original post with details about the overall plan, which has expanded a bit, but I'm encouraged by all the ❤️ so far:
- adding to CRA and RWJS.com repos
- use GH Action to merge all contributor data into this repo (that plus bot might result in AutomationHell™, TBD)

Lastly, the All-contributors project has predetermined specs for the emojis and does not allow custom additions, which is why I'm removing them from the display for now. However, when adding individuals, we'll use a set of three emojis to specific which repo they contributed to. This will be stored in the json data and we can put it to use at a later time if needed/desired:
- 📖 (doc) == Redwoodjs.com
- 🔧 (tool) == Create-Redwood-App
- 💻 (code) == Framework
